src/sources/opentelemetry/log.rs

```diff
@@ -0,0 +1,254 @@
+use crate::{
+    config::{
+        AcknowledgementsConfig, DataType, GenerateConfig, Output, Resource, SourceConfig,
+        SourceContext,
+    },
+    internal_events::{EventsReceived, StreamClosedError},
+    opentelemetry::LogService::{
+        logs_service_server::{LogsService, LogsServiceServer},
+        ExportLogsServiceRequest, ExportLogsServiceResponse,
+    },
+    serde::bool_or_struct,
+    sources::{util::grpc::run_grpc_server, Source},
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+    SourceSender,
+};
+use futures::TryFutureExt;
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use tonic::{Request, Response, Status};
+use vector_core::{
+    config::LogNamespace,
+    event::{BatchNotifier, BatchStatus, BatchStatusReceiver, Event},
+    ByteSizeOf,
+};
+
+pub const LOGS: &str = "logs";
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct OpentelemetryConfig {
+    address: SocketAddr,
+    #[serde(default)]
+    tls: Option<TlsEnableableConfig>,
+    #[serde(default, deserialize_with = "bool_or_struct")]
+    acknowledgements: AcknowledgementsConfig,
+    /// If this setting is set to `true` logs, metrics and traces will be sent to different outputs.
+    ///
+    /// For a source component named `otel` the received logs, metrics, and traces can then be accessed by specifying
+    /// `otel.logs`, `otel.metrics`, and `otel.traces`, respectively, as the input to another component.
+    #[serde(default = "crate::serde::default_false")]
+    multiple_outputs: bool,
```

Comment:
I think we should just drop the `multiple_outputs` and _always_ do use multiple outputs. We added `multiple_outputs` to the `datadog_agent` source to preserve compatibility, but since this is a new source, we can just use the preferred mechanism of multiple outputs for sources that generate multiple types of data.